src/sentry/event_manager.py

```diff
@@ -821,6 +828,23 @@ def _get_or_create_release_many(jobs, projects):
                     pop_tag(job["data"], "dist")
                     set_tag(job["data"], "sentry:dist", job["dist"].name)
 
+            with locks.get(
+                generate_dynamic_sampling_lock_key(project_id),
+                duration=60,
+                name="dynamic-sampling-lock",
+            ).acquire():
```

Comment:
As far as I know, `acquire` is more like a "try acquire" in that it raises an exception when the lock has already been acquired by someone else. How do we handle that case?

https://github.com/getsentry/sentry/blob/5462ee11ad11ebb9a50323befcd286816d7898c8/src/sentry/utils/locking/lock.py#L22-L30

Reply:
oops yeah you are right